main/auth/auth.py

```diff
@@ -132,6 +132,24 @@ def decorated_function(*args, **kwargs):
   return decorated_function
 
 
+def cron_required(f):
+  decorator_order_guard(f, 'auth.cron_required')
+
+  @functools.wraps(f)
+  def decorated_function(*args, **kwargs):
+    if config.PRODUCTION and 'X-Appengine-Cron' in flask.request.headers:
+      return f(*args, **kwargs)
+    if is_logged_in() and current_user_db().admin:
+      return f(*args, **kwargs)
+    if not is_logged_in() and flask.request.path.startswith('/api/'):
+      return flask.abort(401)
+    if not is_logged_in():
+      return flask.redirect(flask.url_for('signin', next=flask.request.url))
+    return flask.abort(403)
+
+  return decorated_function
+
+
```

Comment:
Correct, as said, I like to split the responsibility in this case. Using the following `control/cron.py`:

```
# coding: utf-8

from datetime import datetime
from datetime import timedelta

import auth
import task

from main import app


@app.route('/cron/stats/calc/<string:when>/')
@auth.cron_required
def cron_stats_calc(when=''):
  if when == 'daily':
    ts = datetime.utcnow() + timedelta(-1)
    tasks, msg = task.schedule_stats_calc(ts.year, ts.month, ts.day)
    return msg, 200
  return '', 403
```

Where the `task.schedule_stats_calc` is the shared common worker method with the admin controlled manual calculation endpoints.